pkg/api/steve/supportconfigs/generator.go

```diff
@@ -29,41 +30,54 @@ const (
 	logPrefix           = "support-config-generator"
 )
 
-var errNotImplemented = errors.New("not implemented")
+var errNotFound = errors.New("not implemented")
 
 type GeneratorHandler struct {
 	ConfigMaps           v1.ConfigMapInterface
 	SubjectAccessReviews authv1.SubjectAccessReviewInterface
+	adapterUtil          *cspadapter.ChartUtil
 }
 
 func NewGeneratorHandler(scaledContext *config.ScaledContext) GeneratorHandler {
 	return GeneratorHandler{
 		ConfigMaps:           scaledContext.Core.ConfigMaps(cspAdapterNamespace),
 		SubjectAccessReviews: scaledContext.K8sClient.AuthorizationV1().SubjectAccessReviews(),
+		adapterUtil:          cspadapter.NewChartUtil(scaledContext.Wrangler.RESTClientGetter),
 	}
 }
 
 func (h *GeneratorHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 	authorized, err := h.authorize(request)
 	if err != nil {
-		util.ReturnHTTPError(writer, request, http.StatusForbidden, "forbidden")
+		util.ReturnHTTPError(writer, request, http.StatusForbidden, http.StatusText(http.StatusForbidden))
 		logrus.Errorf("[%s] Failed to authorize user with error: %s", logPrefix, err.Error())
 		return
 	}
 	if !authorized {
-		util.ReturnHTTPError(writer, request, http.StatusForbidden, "forbidden")
+		util.ReturnHTTPError(writer, request, http.StatusForbidden, http.StatusText(http.StatusForbidden))
+		return
+	}
+	_, err = h.adapterUtil.GetRelease()
+	if err != nil {
+		if errors.Is(err, cspadapter.ErrNotFound) {
+			// if the adapter is not installed, return a 501 so the user knows to install the adapter
+			util.ReturnHTTPError(writer, request, http.StatusNotImplemented, "csp-adapter must be installed in order to to generate supportconfigs")
+			return
+		}
+		logrus.Warnf("[%s] Error when attempting to determine if adapter is installed, %s", logPrefix, err)
+		util.ReturnHTTPError(writer, request, http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError))
 		return
 	}
 	logrus.Infof("[%s] Generating supportconfig", logPrefix)
 	archive, err := h.generateSupportConfig()
 	logrus.Infof("[%s] Done Generating supportconfig", logPrefix)
 	if err != nil {
-		if errors.Is(err, errNotImplemented) {
-			util.ReturnHTTPError(writer, request, http.StatusNotImplemented, "csp-adapter must be installed in order to to generate supportconfigs")
+		if errors.Is(err, errNotFound) {
+			util.ReturnHTTPError(writer, request, http.StatusServiceUnavailable, "supportconfig not yet generated, try again later")
 			return
 		}
 		logrus.Errorf("[%s] Error when generating supportconfig %v", logPrefix, err)
-		util.ReturnHTTPError(writer, request, http.StatusInternalServerError, "Internal Server Error")
+		util.ReturnHTTPError(writer, request, http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError))
 		return
 	}
 	writer.Header().Set("Content-Type", tarContentType)
```

Comment:
This comment is for line 85, I can't leave a line comment because of this [issue](https://github.com/isaacs/github/issues/1655) with Github. 

The error return from `Copy` is not being handled.
```golang
n, err := io.Copy(writer, archive)
```

Reply:
Good call, will fix.